### PR TITLE
[cmake] use precompiled headers.

### PR DESCRIPTION
**Pull request by dlyr**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature: use cmake precompiled header mecanism.

* **What is the current behavior?** (You can also link to an open issue here)
All system + external headers are parsed each time included.

* **What is the new behavior (if this is a feature change)?**

Add precompiled header to lib targets using system and external headers.
Also each lib uses precompiled headers for the other radium lib which it depends from.
Hence headers are parsed and precompiled once for each radium lib target.

I have added a script to count how many time include file are included, and arbitrary keep include included 3 times or more.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
Nope.

### CODE REVIEW

File: src/IO/CMakeLists.txt [hunk 1]
```diff
@@ -65,4 +65,23 @@ configure_radium_library(
...
+    <assimp/matrix4x4.h>
+    <assimp/quaternion.h>
+    <assimp/types.h>
+    <assimp/vector3.h>
+    <fstream>
+    <map>
+    <memory>
+    <set>
+    <string>
+)
```
Comment (nmellado) [inline src/IO/CMakeLists.txt:83]: Duplicate ?

Comment (dlyr) [reply src/IO/CMakeLists.txt:83]: yes

File: src/Core/CMakeLists.txt [hunk 2]
```diff
@@ -87,4 +87,6 @@ configure_radium_library(
 )
 set(RADIUM_COMPONENTS ${RADIUM_COMPONENTS} ${ra_core_target} PARENT_SCOPE)
 
+target_precompile_headers(${ra_core_target} PRIVATE pch.hpp)
```
[the remark this answers is not in the document]
Comment (dlyr) [reply src/Core/CMakeLists.txt:90]: nope, see cmake doc.
"Header file names specified with angle brackets (e.g. <unordered_map>) or explicit double quotes (escaped for the [cmake-language(7)](https://cmake.org/cmake/help/latest/manual/cmake-language.7.html#manual:cmake-language(7)), e.g. [["other_header.h"]]) will be treated as is, and include directories must be available for the compiler to find them. Other header file names (e.g. project_header.h) are interpreted as being relative to the current source directory (e.g. [CMAKE_CURRENT_SOURCE_DIR](https://cmake.org/cmake/help/latest/variable/CMAKE_CURRENT_SOURCE_DIR.html#variable:CMAKE_CURRENT_SOURCE_DIR)) and will be included by absolute path. For example:"

But we can add full path if clearer.

File: src/Gui/pch.hpp [hunk 3]
```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <glbinding/gl/gl.h>
+#include <glbinding/gl45core/gl.h>
+#include <glbinding/gl45ext/gl.h>
+
```
[the remark this answers is not in the document]
Comment (dlyr) [reply src/Gui/pch.hpp:6]: this was more or less a test to check Qt/glbinding interaction. I remove it,

**Comment by dlyr** [comment]
On my computer, full build time reduce by 22%.

I have also compiled file one by one (I mean touch one file and rebuild all) for each file, with and without pch.
The fact that some files from Radium are in pch (e.g. `Core/Resources/Resources.hpp` in `src/Engine/pch.hpp`) makes their compile time quite high (because a modification in such files trigger the recompilation of more files (e.g. whole libEngine).

Here is the graph of gain/loss
Screenshot from 2022-07-11 08-14-59

**Review by MathiasPaulin — APPROVED** [review]

**Pull request merged**